hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -38,39 +39,108 @@
 public interface DefinitionsService {
 
     /*
-        CRUD interface for Trigger
+        CRUD interface for Triggers
      */
 
     /**
-     * Create a new <code>Trigger</code>.  <code>Conditions</code> and <code>Actions</code> are manipulated in separate
-     * calls. The new <code>Trigger</code> will be persisted.  When fully defined a call to
-     * {@link #updateTrigger(String, Trigger)}
-     * is needed to enable the <code>Trigger</code>.
+     * Create a new <code>Trigger</code>. <code>Conditions</code> and <code>Dampening</code> are
+     * manipulated in separate calls. The new <code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(String, Trigger)} is needed to enable the <code>Trigger</code>.
+     * <p>
+     * A non-group trigger can never be made into a group trigger, and vice-versa.
+     * </p>
      * @param tenantId Tenant where trigger is created
      * @param trigger New trigger definition to be added
      * @throws Exception If the <code>Trigger</code> already exists.
+     * @see {@link #addGroupTrigger(String, Trigger)} for adding a group trigger.
      */
     void addTrigger(String tenantId, Trigger trigger) throws Exception;
 
     /**
+     * Create a new Group <code>Trigger</code>. <code>Conditions</code> and <code>Dampening</code> are
+     * manipulated in separate calls. The new <code>Group Trigger</code> will be persisted.  When fully
+     * defined a call to {@link #updateGroupTrigger(String, Trigger)} is needed to enable the <code>Trigger</code>.
+     * <p>
+     * A non-group trigger can never be made into a group trigger, and vice-versa.
+     * </p>
+     * @param tenantId Tenant where trigger is created
+     * @param groupTrigger New trigger definition to be added
+     * @throws Exception If the <code>Trigger</code> already exists.
+     * @see {@link #addTrigger(String, Trigger)} for adding a non-group trigger.
+     */
+    void addGroupTrigger(String tenantId, Trigger groupTrigger) throws Exception;
+
+    /**
+     * Generate a member trigger for the specified group trigger. The dataIdMap replaces the tokens in the
+     * group trigger's conditions with actual dataIds. The member trigger gets the enabled state of the group.
```

Comment:
As an enhancement, I think that an example of tokens of dataIdMap javadoc can be convenient.
I remember the learning cuve that UI team had first time with the API and a couple of examples can help here.
Nothing complex, just to add a couple of lines of what is expected for dataIdMap a token is also here in the API.